packages/kit/src/runtime/server/index.js

```diff
@@ -151,6 +151,38 @@ export async function respond(request, options, state = {}) {
 					event.url = new URL(event.url.origin + normalized + event.url.search);
 				}
 
+				const requestRouteKey = request.headers.get('x-sveltekit-load');
+				if (requestRouteKey !== null) {
+					const route = options.manifest._.routes.find(
+						(route) => route.type === 'page' && route.key === requestRouteKey
+					);
+					if (route) {
+						// @ts-ignore
+						const shadow = route.shadow;
+						if (shadow) {
+							const match = route.pattern.exec(decoded);
+							if (match) {
+								event.params = route.params ? decode_params(route.params(match)) : {};
+								const response = await render_endpoint(event, await shadow());
+								if (response) {
+									if (response.status >= 300 && response.status < 400) {
+										const location = response.headers.get('location');
+										if (location) {
+											const headers = new Headers(response.headers);
+											headers.set('x-sveltekit-location', location);
+											return new Response(undefined, {
+												status: 204,
+												headers
+											});
+										}
+									}
+									return response;
+								}
+							}
+						}
+					}
+				}
+
```

Comment:
Could this not be replaced with something along these lines inside the `for` loop below?

```js
if (key && route.key !== key) continue;
```

Reply:
Yes, it's better.